test/syscalls/linux/alloc_fd_benchmark.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2019 Ant Financial Services Group.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "benchmark/benchmark.h"
+#include "gtest/gtest.h"
+
+namespace gvisor {
+namespace testing {
+
+namespace {
+
+const int DEFAULT_FILE_NUM = 10000;
+
+void BM_open_serial(benchmark::State &state) {
+  auto fd_table = std::vector<int>(DEFAULT_FILE_NUM);
+  for (auto _ : state) {
+    for (int i = 0; i < DEFAULT_FILE_NUM; ++i) {
+      fd_table[i] = memfd_create(std::to_string(i).c_str(), MFD_CLOEXEC);
+      close(fd_table[i]);
+    }
+  }
+}
+BENCHMARK(BM_open_serial);
+
+void BM_open_zigzag(benchmark::State &state) {
+  auto fd_table = std::vector<int>(DEFAULT_FILE_NUM);
+  for (auto _ : state) {
+    for (int i = 0; i < DEFAULT_FILE_NUM; ++i) {
+      fd_table[i] = memfd_create(std::to_string(i).c_str(), MFD_CLOEXEC);
+    }
+    for (int i = DEFAULT_FILE_NUM; i >= 0; --i) {
+      close(fd_table[i]);
+    }
+  }
+}
+BENCHMARK(BM_open_zigzag);
+
+void BM_open_one_by_one(benchmark::State &state) {
+  auto fd_table = std::vector<int>(DEFAULT_FILE_NUM);
+  for (auto _ : state) {
+    for (int i = 0; i < DEFAULT_FILE_NUM; ++i) {
+      fd_table[i] = memfd_create(std::to_string(i).c_str(), MFD_CLOEXEC);
+      close(fd_table[i]);
+    }
+  }
+}
+BENCHMARK(BM_open_one_by_one);
+
+void BM_open_random(benchmark::State &state) {
+  // Init random access vector
+  std::vector<int> position_of_positive_number(DEFAULT_FILE_NUM + 1, -1);
+
```

Comment:
negative

Reply:
delete position_of_positive_number in newer edition.